gui/common/functions_utility~autociv.js

```diff
@@ -124,3 +124,21 @@ function brightenedColor(color)
     autociv_ColorsSeenBefore[color] = hslToRgb(h, s, Math.min(l + 0.3, 0.6)).join(" ");
     return autociv_ColorsSeenBefore[color];
 }
+
+const autociv_NameSeenBefore = {};
+
+/**
+ * Truncates a string if it is longer than the specified maximum string length.
+ * @param   {string}  str  [str description]
+ * @param   {number}  num  [num description]
+ * @return  {string}       Truncated string with a horizontal ellipsis
+ */
+function truncateString(str, num)
+{
+    if (!autociv_NameSeenBefore[str])
+    {
+        let {nick} = splitRatingFromNick(str)
+        autociv_NameSeenBefore[str] = nick.length > num ? nick.slice(0, num-1) + "…" : nick;
+    }
+    return autociv_NameSeenBefore[str]
+}
```

Comment:
Move all this inside StatsOverlay class that is /session, no need to have it in /common